modules/apps/transfer/keeper/relay.go

```diff
@@ -115,7 +115,7 @@ func (k Keeper) sendTransfer(
 		// chain inside the packet data. The receiving chain will perform denom
 		// prefixing as necessary.
 
-		if token.Denom.SenderChainIsSource(sourcePort, sourceChannel) {
+		if !token.Denom.HasPrefix(sourcePort, sourceChannel) {
```

Comment:
I think if we do this change, we should flip flop the if statement. I think there's a bit of unnecessary mental gymnastics to figure out what a non prefix set means. I feel like it's easier to write a comment which states:
```go
// if the port and channel we are sending the token on was prefixed to this denom
// then we must be returning the token back to the chain they originated from
```
then I think the burn coins logic is more natural